cuda/test/factorization/par_ilu_kernels.cpp

```diff
@@ -176,6 +220,63 @@ class ParIlu : public ::testing::Test {
 };
 
 
+TEST_F(ParIlu, CudaKernelAddDiagonalElementsSortedEquivalentToRef)
+{
+    index_type num_rows{600};
+    index_type num_cols{600};
+    auto mtx_ref = gen_mtx<Csr>(num_rows, num_cols);
+    auto mtx_cuda = Csr::create(cuda);
+    mtx_cuda->copy_from(gko::lend(mtx_ref));
+
+    gko::kernels::reference::par_ilu_factorization::add_diagonal_elements(
+        ref, gko::lend(mtx_ref), true);
+    gko::kernels::cuda::par_ilu_factorization::add_diagonal_elements(
+        cuda, gko::lend(mtx_cuda), true);
+
+    ASSERT_TRUE(mtx_ref->is_sorted_by_column_index());
+    GKO_ASSERT_MTX_NEAR(mtx_ref, mtx_cuda, 0.);
+    GKO_ASSERT_MTX_EQ_SPARSITY(mtx_ref, mtx_cuda);
+}
+
+
+TEST_F(ParIlu, CudaKernelAddDiagonalElementsUnsortedEquivalentToRef)
+{
+    index_type num_rows{600};
+    index_type num_cols{600};
+    auto mtx_ref = gen_unsorted_mtx(num_rows, num_cols);
+    auto mtx_cuda = Csr::create(cuda);
+    mtx_cuda->copy_from(gko::lend(mtx_ref));
+
+    gko::kernels::reference::par_ilu_factorization::add_diagonal_elements(
+        ref, gko::lend(mtx_ref), false);
+    gko::kernels::cuda::par_ilu_factorization::add_diagonal_elements(
+        cuda, gko::lend(mtx_cuda), false);
+
+    ASSERT_FALSE(mtx_ref->is_sorted_by_column_index());
+    GKO_ASSERT_MTX_NEAR(mtx_ref, mtx_cuda, 0.);
+    GKO_ASSERT_MTX_EQ_SPARSITY(mtx_ref, mtx_cuda);
+}
+
+
+TEST_F(ParIlu, CudaKernelAddDiagonalElementsAsymetricEquivalentToRef)
```

Comment:
I think you mean non-square ? Square matrices can be asymmetric.

Reply:
That is true, my bad.